docs/proposals/20191030-machine-health-checking.md

```diff
@@ -164,10 +165,10 @@ type target struct {
 
 - Calculate the number of unhealthy targets.
 - Compare current number against `maxUnhealthy` threshold and temporary short circuits remediation if the threshold is met.
-- Annotates unhealthy target machines as described above.
+- Marks unhealthy target machines with the conditions as described above.
 
 Out of band:
-- The owning controller observes the annotation and is responsible to remediate the machine.
+- The owning controller observes the OwnerRemediationSucceeded=False condition and is responsible to remediate the machine.
 - The owning controller performs any pre-deletion tasks required.
```

Comment:
I know is not introduced by this PR but could we elaborate on what are `pre-deletion tasks`?

To make sure we don't miss any gap and illustrate better how and when the conditions are expected to interact, could we please flesh out here the details and  workflow for each of the remediation processes owned by capi here, i.e regular vs controlPlane machines?

Reply:
We should probably remove that goal, it's not relevant or in scope for this proposal

Reply:
For the sake of the contract all that matters is that the owning controller handles the conditions. I don't even think it's a requirement anymore that we delete the machine, since this proposal introduces a mechanism for marking it as successfully remediated.